dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/audit/legacy/DefaultAuditObjectFactory.java

```diff
@@ -46,9 +59,24 @@
 {
     private final RenderService renderService;
 
-    public DefaultAuditObjectFactory( RenderService renderService )
+    private final CacheProvider cacheProvider;
+
+    private Cache<Map<Field, Method>> cachedAuditAttributeFields;
+
+    public DefaultAuditObjectFactory( RenderService renderService, CacheProvider cacheProvider )
     {
         this.renderService = renderService;
+        this.cacheProvider = cacheProvider;
+    }
+
+    @PostConstruct
+    public void initCache()
+    {
+        cachedAuditAttributeFields = cacheProvider.newCacheBuilder( Field.class, Method.class )
+            .forRegion( "auditAttributeFields" )
+            .withInitialCapacity( 100 )
+            .withMaximumSize( 500 )
```

Comment:
We should probably keep this under watch and make sure we don't kick any fields out... we have quite a lot of classes, and while I'm not sure how many places we will add this attribute, we should never be in a situation where some fields are kicked out of the cache

Reply:
increased it to 500-1000